packages/flutter_tools/lib/src/base/flags.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:args/args.dart';
+
+import 'context.dart';
+
+Flags get flags => context?.getVariable(Flags) ?? const _EmptyFlags();
+
+class Flags {
+  Flags(this._globalResults) {
+    assert(_globalResults != null);
+  }
+
+  final ArgResults _globalResults;
+
+  dynamic operator [](String key) {
+    final ArgResults commandResults = _globalResults.command;
+    if (commandResults?.options?.contains(key) ?? false) {
+      return commandResults[key];
+    } else if (_globalResults.options.contains(key)) {
```

Comment:
If `commandResults.options` can be `null`, why can't `_globalResults.options`?

Reply:
`commandResults.options` can't be `null`, but `commandResults` can be `null`, which can make `commandResults?.options` `null`, which necessitates the second `?.`